mock/src/components/Commands.ts

```diff
@@ -0,0 +1,106 @@
+import { REPLFunction } from "./REPLFunction";
+import { Dispatch, SetStateAction, useState } from "react";
+import { mockedDataMap, mockedResultsMap } from "./mockedJson";
+import { HistoryEntry } from "./REPL";
+
+// An export that REPLInput can use to map certain commands to functions
+export const commandMap = new Map<string, REPLFunction>();
+
+// Create a varaible to store the file
+let loadedData: string[][];
+let file: { data: string[][]; hasHeaders: boolean } | undefined;
+
+export function addCommand(
+  commandName: string,
+  commandFunction: REPLFunction
+): void {
+  commandMap.set(commandName, commandFunction);
+}
+
+export function removeCommand(commandName: string): void {
+  commandMap.delete(commandName);
+}
+
+// Initialize the basic commands
+export function setCommandMap() {
```

Comment:
Using a map is an awesome idea. It makes the code easy to edit and add on new commands. One thing my group did that might help with readability is splitting the map and all the individual command functions into two files to decrease clutter and improve delegation!